packages/git/src/browser/git-decoration-provider.ts

```diff
@@ -14,54 +14,103 @@
 // SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
 // *****************************************************************************
 
-import { inject, injectable } from '@theia/core/shared/inversify';
+import { inject, injectable, postConstruct } from '@theia/core/shared/inversify';
 import { GitFileChange, GitFileStatus, GitStatusChangeEvent } from '../common';
 import { CancellationToken, Emitter, Event } from '@theia/core/lib/common';
 import { Decoration, DecorationsProvider } from '@theia/core/lib/browser/decorations-service';
 import { GitRepositoryTracker } from './git-repository-tracker';
 import URI from '@theia/core/lib/common/uri';
+import { GitConfiguration, GitPreferences } from './git-preferences';
+import { PreferenceChangeEvent } from '@theia/core/lib/browser';
 
 @injectable()
 export class GitDecorationProvider implements DecorationsProvider {
 
+    @inject(GitPreferences) protected readonly preferences: GitPreferences;
+    @inject(GitRepositoryTracker) protected readonly gitRepositoryTracker: GitRepositoryTracker;
+
+    protected decorationsEnabled: boolean;
+    protected colorsEnabled: boolean;
+
+    protected decorations = new Map<string, Decoration>();
+    protected uris: Set<string> = new Set<string>();
+
+    /**
+     * Cached change event for re-rendering decorations.
+     */
+    protected changeEvent: GitStatusChangeEvent | undefined;
+
     private readonly onDidChangeDecorationsEmitter = new Emitter<URI[]>();
     readonly onDidChange: Event<URI[]> = this.onDidChangeDecorationsEmitter.event;
 
-    private decorations = new Map<string, Decoration>();
+    @postConstruct()
+    protected init(): void {
+        this.decorationsEnabled = this.preferences['git.decorations.enabled'];
+        this.colorsEnabled = this.preferences['git.decorations.colors'];
+        this.gitRepositoryTracker.onGitEvent((event: GitStatusChangeEvent | undefined) => this.handleGitEvent(event));
```

Comment:
Minor: Do we need to be this explicit?
```suggestion
        this.gitRepositoryTracker.onGitEvent(event => this.handleGitEvent(event));
```